feat: update to angular 12

jafin opened the pull request:
* Update component to Angular 12
* Replace TSLint with ESLint

jafin committed:
Run ng update to 12

jafin committed:
Update validation package dependencies

jafin committed:
fix Validator example.

jafin committed:
style demo with cirrus

jafin committed:
align rxjs version to angular 12 (6.6.7)

jafin committed:
task (ESLint): Migrate TSLint to ESLint

jafin committed:
tidy: remove typings that are in @types/node

davidwalschots committed:
use node 14

davidwalschots commented on .gitignore:
```diff
@@ -44,3 +44,4 @@ testem.log
 Thumbs.db
 
 TESTS-*.xml
+/angular-reactive-validation/node_modules/*
```
**thought**: Probably you ran `npm install` within `/angular-reactive-validation`. There's no need to do so. Run `npm install` in the root and remove the `node_modules` folder in the subfolder. Everything should still work and this line can be removed again 😀.

jafin replied:
Yes i did. Fixed.

davidwalschots commented on .eslintrc.json:
```diff
@@ -0,0 +1,68 @@
+{
```
Thanks for migrating to eslint. I'd like to note that on opening various ts files within vscode I'm seeing a lot of eslint errors popping up.
**suggestion**: Go through each individual ts file and apply the rules.

jafin replied:
Ok fixed, 1) I didn't see this I was in Webstorm. and 2) running `ng lint` did not pick up any lint warnings.. Perhaps something is still missing as I would of expected ng lint to display warnings you were seeing?

davidwalschots replied:
I've included additional paths for linting in `angular.json`. It feels somewhat odd to have to define more paths while within the `"projects" -> "angular-reactive-validation"` scope. That could be related to the fact that when I introduced the project in a project, there was no true angular support for doing so yet.

davidwalschots commented on angular-reactive-validation/package.json:
```diff
@@ -20,18 +20,18 @@
-    "@angular/common": "^11.0.0",
-    "@angular/forms": "^11.0.0",
-    "rxjs": "^6.5.4"
+    "@angular/core": "^12.2.15",
+    "@angular/common": "^12.2.15",
+    "@angular/forms": "^12.2.15",
+    "rxjs": "^6.6.7"
```
**thought**: The package can be compatible with any Angular 12.x version by changing the `peerDependencies` to be the lowest possible version we want to be compatible with. In this case `"^12.0.0"` for Angular and `"^6.5.3"` for rxjs (so can actually be lower than the previous version set there, oops).

jafin replied:
done

davidwalschots commented on angular-reactive-validation/package-lock.json:
```diff
@@ -0,0 +1,252 @@
+{
```
As noted earlier, you probably ran `npm install` in the sub directory, causing this file to be created.
**suggestion**: Remove the file.

jafin replied:
Fixed. removed.

davidwalschots commented on src/app/app.component.html:
```diff
@@ -1,27 +1,44 @@
+        </arv-validation-message>
+      </arv-validation-messages>
+    </div>
+    <div>
```
Textual alignment of this div and some of its content is a bit wonky. 😀

davidwalschots reviewed (changes requested):
Thank you for your contribution! I have taken the liberty to modify the node version the build pipeline uses to v14. Some small remarks here and there. Looking forward to getting this merged!

Do you by any chance plan on updating to Angular 13 right after this PR?

jafin committed:
-fix ESlint warnings.

-Fix gitignore,
-set package-lock back to 12.0.0
-fix indenting on app.component

jafin commented:
> Do you by any chance plan on updating to Angular 13 right after this PR?

No actual plan to go to v13 at the moment for me.

davidwalschots committed:
lint all project files

davidwalschots committed:
lint in pipeline

davidwalschots committed:
bump version

davidwalschots committed:
remove ban-types eslint disable

jafin commented:
@davidwalschots thanks for fixing up the PR!

davidwalschots commented:
Thanks for your contribution!
